packages/monaco/src/browser/monaco-editor-model.ts

```diff
@@ -273,11 +296,22 @@ export class MonacoEditorModel implements ITextEditorModel, TextEditorDocument {
         return this.resource.saveContents === undefined;
     }
 
-    get onDispose(): monaco.IEvent<void> {
+    isReadonly(): boolean {
+        return this.readOnly;
+    }
+
+    get onDispose(): Monaco.IEvent<void> {
+        return this.toDispose.onDispose;
+    }
+
+    get onWillDispose(): Event<void> {
         return this.toDispose.onDispose;
     }
 
-    get textEditorModel(): monaco.editor.IModel {
+    // We have a TypeScript problem here. There is a const enum `DefaultEndOfLine` used for ITextModel and a non-const redeclaration of that enum in the public API in
+    // Monaco.editor. The values will be the same, but TS won't accept that the two enums are equivalent, so it says these types are irreconcilable.
+    get textEditorModel(): Monaco.editor.ITextModel & ITextModel {
+        // @ts-expect-error ts(2322)
```

Comment:
Can we maybe fix some of the monaco.d.ts types using tricks like https://www.typescriptlang.org/docs/handbook/2/mapped-types.html?

Reply:
I don't think it would be a general solution, since we would need to say something like 'for every field, check if it's a const enum, and if it is, replace it with the equivalent (but TS doesn't believe they're equivalent) non-const enum from this file.' I don't think either 'check if it's a const enum' or 'replace with a dynamic lookup of the equivalent from the public API (if there is one)' parts of that are available, but it may be possible to make some sort of compatibility hacks for individual cases, if that's preferable to typecasts / ts-expect in the code.